### ISSUE TITLE
commit fails with "error determining uncompressed digest for blob" when using vfs

### ISSUE THREAD

**Issue by giuseppe** [projectatomic/buildah#502]
I've tried to build an image using the vfs storage driver (that could be used from an user namespace).

**Steps to reproduce the issue:**
```
ctr1=`buildah --storage-driver vfs from --pull fedora:27`
buildah --storage-driver vfs run $ctr1 -- dnf  upgrade -y
buildah --storage-driver vfs run $ctr1 -- dnf install -y lighttpd
buildah --storage-driver vfs config $ctr1 --annotation "com.example.build.host=fedora-27"
buildah --storage-driver vfs config $ctr1 --cmd "/usr/sbin/lighttpd -D -f /etc/lighttpd/lighttpd.conf"
buildah --storage-driver vfs config $ctr1 --port 80
buildah --storage-driver vfs commit $ctr1 giuseppe/lighttpd
```

**Describe the results you received:**
```
error committing container "fedora-working-container-7" to "giuseppe/lighttpd": error copying layer and metadata: error committing new image "containers-storage:[vfs@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]docker.io/giuseppe/lighttpd:latest": error determining uncompressed digest for blob "sha256:c8be87302b4b2a1c84a22a29f43e6b45ca68dff1909f2cce967f372e7ed28b49"
```

**Describe the results you expected:**

"buildah commit" succeded

**Output of `rpm -q buildah` or `apt list buildah`:**

using buildah from upstream, commit cb42905a7f5e91a4407b090947cd0ea0d375ada0

**Output of `cat /etc/*release`:**

```
# cat /etc/os-release 
NAME=Fedora
VERSION="27 (Workstation Edition)"
ID=fedora
VERSION_ID=27
PRETTY_NAME="Fedora 27 (Workstation Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:27"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=27
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=27
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation

```

**Output of `uname -a`:**
```
# uname -a
Linux helium 4.14.16-300.fc27.x86_64 #1 SMP Wed Jan 31 19:24:27 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# cat /etc/containers/storage.conf
# storage.conf is the configuration file for all tools
# that share the containers/storage libraries
# See man 5 containers-storage.conf for more information

# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"
```

**Comment by rhatdan** [comment]
@mtrmac @nalind PTAL

**Comment by nalind** [comment]
I can't reproduce this when starting with a clean pull of the fedora:27 package.  Have any tools which were built with an older version of the storage library been used to modify the storage tree?  Do you have the output from using the `--debug` flag with `buildah commit`?

**Comment by nalind** [comment]
Ah, I can reproduce this.

**Comment by rhatdan** [comment]
Where does this fix for this need to be done?

**Comment by nalind** [comment]
This should be fixed by #506.

**Comment by nalind** [comment]
Can we recheck with a version that's later than when we merged #508?

**Comment by giuseppe** [comment]
this issue is fixed
